### PR TITLE
streamingccl: Ensure appropriate privileges when replicating.

### PR DESCRIPTION
**Pull request by miretskiy**
Add checks to replication stream manager to ensure appropriate privileges
and require enterprise license when executing streaming replication.

Release Notes: None

### CODE REVIEW

**Review by gh-casper — APPROVED** [review]

File: pkg/ccl/streamingccl/streamproducer/replication_manager.go [hunk 1]
```diff
@@ -45,14 +49,36 @@ func (r replicationStreamManagerImpl) UpdateReplicationStreamProgress(
...
-		return &replicationStreamManagerImpl{}, nil
+func newReplicationStreamManagerWithPrivilegesCheck(
+	evalCtx *tree.EvalContext,
+) (streaming.ReplicationStreamManager, error) {
+	isAdmin, err := evalCtx.SessionAccessor.HasAdminRole(evalCtx.Context)
+	if err != nil {
+		return nil, err
+	}
+
+	if !isAdmin && !evalCtx.SessionData().User().IsRootUser() {
```
Comment (dt) [inline pkg/ccl/streamingccl/streamproducer/replication_manager.go:66]: isn't admin role enough here? if being root good enough if root isn't an admin anymore?

Comment (miretskiy) [reply pkg/ccl/streamingccl/streamproducer/replication_manager.go:66]: Ack.

**Pull request merged**